### PR TITLE
Review

### PR DESCRIPTION
**Pull request by lrmi0808**
Review model

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -6,8 +6,10 @@ def index
   end
 
   def show
+    @user = User.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/courses_controller.rb:9]: You mean:
```suggestion
    @user = User.find_by(id: params[:user_id])
```

File: app/controllers/registers_controller.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ def new
 
   def create
     @user = User.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/registers_controller.rb:9]: you mean:
```suggestion
    @user = User.find_by(id: params[:user_id])
```

File: app/controllers/registers_controller.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@ def new
...
-    @course = Course.find_by(params[:id])
+    @course = Course.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/registers_controller.rb:10]:
```suggestion
    @course = Course.find_by(id: params[:course_id])
```

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+class ReviewsController < ApplicationController
+  before_action :logged_in_user
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    @course = Course.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/reviews_controller.rb:10]:
```suggestion
    @course = Course.find_by(id: params[:course_id])
```

File: app/controllers/reviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    if @review.save
+      flash[:success] = "Review created!"
+      redirect_back(fallback_location: @user)
```
Comment (huongnt-2545) [inline app/controllers/reviews_controller.rb:14]: Where do you get `@user` value??

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -9,7 +9,9 @@ def index
 
   def show
     @user = User.find_by(id: params[:id])
+    @course = Course.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:12]: is this correct? why do you find course by `params[:id]` value from here?

File: app/views/courses/show.html.erb [hunk 7]
```diff
@@ -73,4 +73,34 @@
       </div>
     </div>
   </div>
+
+  <div>
+    <div>
+      <h2><%= t(".review") %>:</h2>
+      <% @course.reviews.each do |review| %>
+          <p>
```
Comment (huongnt-2545) [inline app/views/courses/show.html.erb:81]: wrong indent

File: app/views/users/show.html.erb [hunk 8]
```diff
@@ -62,4 +62,27 @@
...
+          </p>
+          <p>
+            <strong><%= t(".review") %>:</strong>
+            <%= review.content %>
+          </p>
+          <p>
+            <% if current_user? @user %>
+              <%= link_to t(".edit"), edit_review_path(review) %>
+              <%= link_to t(".delete"), review_path(review), class:"btn btn-mini",
+                    method: :delete,
```
Comment (huongnt-2545) [inline app/views/users/show.html.erb:82]: wrong indent

**Comment by lrmi0808** [comment]
Ready

File: app/views/courses/show.html.erb [hunk 9]
```diff
@@ -10,7 +10,7 @@
 
     <% if current_user.admin? %>
       <ul>
-        <li><%= link_to t(".edit"), edit_course_path(@course) %></li>
```
Comment (huongnt-2545) [inline app/views/courses/show.html.erb:13]: keep this

File: app/views/reviews/_form.html.erb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+<%= form_for @review  do |f| %>
+  <%= render "shared/error_messages", object: @review %>
+  <%= f.hidden_field :user_id, :value => current_user.id %>
```
Comment (huongnt-2545) [inline app/views/reviews/_form.html.erb:3]:
```suggestion
  <%= f.hidden_field :user_id, value: current_user.id %>
  <%= f.hidden_field :course_id, value: @course.id %>
```

File: app/views/courses/show.html.erb [hunk 11]
```diff
@@ -73,4 +73,33 @@
...
+          <strong><%= t(".review") %>:</strong>
+          <%= review.content %>
+        </p>
+        <p>
+          <% if current_user.id == review.user_id %>
+            <%= link_to t(".edit"), edit_review_path(review) %>
+          <% end %>
+          <% if current_user.admin? || (current_user.id == review.user_id) %>
+            <%= link_to t(".delete"), review_path(review), class:"btn btn-mini",
+                          method: :delete,
```
Comment (huongnt-2545) [inline app/views/courses/show.html.erb:95]: wrong indent

**Pull request merged**